inspire_mitmproxy/services/base_service.py

```diff
@@ -55,9 +55,14 @@ def handles_request(self, request: MITMRequest) -> bool:
 
         return host in self.hosts_list
 
-    def _get_first_matching_interaction(self, request):
+    def _can_replay(self, interaction: Interaction) -> bool:
```

Comment:
Technicaly, if the `max_replay: 0` or `1` then it becomes "should play" rather than "replay"

Reply:
Also regarding my comment in the integration test, you could instead add a few unit tests to check the corner cases here?